File: app/src/main/java/com/atom/vpn/demo/adapter/CountryAdapter.java
/*
 * Copyright (C) 2018 The Android Open Source Project
 *
 */

package com.atom.vpn.demo.adapter;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.atom.sdk.android.data.model.countries.Country;


public class CountryAdapter extends ArrayAdapter<Country> {


    private Country[] countries;

    public CountryAdapter(Context context, int textViewResourceId,
                          Country[] countries) {
        super(context, textViewResourceId, countries);
        this.countries = countries;
    }

    @Override
    public int getCount(){
        if(countries!=null){
        return countries.length;
        }
        return 0;
    }

    @Override
    public Country getItem(int position){
        return countries[position];
    }

    @Override
    public long getItemId(int position){
        return position;
    }



    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        TextView label = (TextView) super.getView(position, convertView, parent);
        label.setTextColor(Color.BLACK);
        label.setText(countries[position].getName());
        return label;
    }

    @Override
    public View getDropDownView(int position, View convertView,
                                @NonNull ViewGroup parent) {
        TextView label = (TextView) super.getDropDownView(position, convertView, parent);
        label.setTextColor(Color.BLACK);
        label.setText(countries[position].getName());

        return label;
    }
}

File: app/src/main/java/com/atom/vpn/demo/adapter/CountryAdapter.kt
/*
 * Copyright (C) 2018 The Android Open Source Project
 *
 */

package com.atom.vpn.demo.adapter

import android.content.Context
import android.graphics.Color
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.atom.core.models.Country

class CountryAdapter(
    context: Context,
    textViewResourceId: Int,
    private val countries: Array<Country>
) : ArrayAdapter<Country>(context, textViewResourceId, countries) {

    override fun getCount(): Int = countries.size

    override fun getItem(position: Int): Country = countries[position]

    override fun getItemId(position: Int): Long = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val label = super.getView(position, convertView, parent) as TextView
        label.setTextColor(Color.BLACK)
        label.text = countries[position].name
        return label
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        val label = super.getDropDownView(position, convertView, parent) as TextView
        label.setTextColor(Color.BLACK)
        label.text = countries[position].name
        return label
    }
}
